Cargo.toml

```diff
@@ -1,4 +1,5 @@
 [workspace]
+resolver = "2"
 
 members = [
```

Comment:
We can move `lightning-custom-message` crate here now. It was excluded because of the edition, IIRC. Probably could fold it into the `lightning` crate, too, but that may be better saved for a separate refactor. It's not listed on lightningdevkit.org at the moment.